### DISCUSSION TITLE
Rework the model settings

### DISCUSSION THREAD

**Discussion by vbandi** [cline/cline#383] [Feature Requests]
# Goal: 
Some tasks only need a quick and smaller model, others work better with OpenAI or Claude. 

I'd like to be able to dynamically switch between different models and providers as needed, even during tasks. I'd also like to be able to use multiple models from the same provider. This would also help circumventing the daily quota limitations of some providers - we wouldn't spend precious Claude tokens on simple tasks.

# Plan
Currently, Claude Dev has the ability to speak to multiple providers. However, it is still essentially a single-model system.

I'd like to be able to define multiple models. These would be defined in a UI similar to the current settings (including provider, model selection, secrets, etc), but we'd need more than one. Every model / provider would have a name the user can set. 

In the chat entry textbox, there should be a small dropdown (maybe in the lower right corner, like in Cursor) that I can use to select any one of the models defined in the settings for the conversation turn. I should be able to switch between these models at any time during the conversation, so a Task can start with Claude Sonnet, and end with OpenAI's GPT4o.

**Comment by JohannesKoerner** [comment]
Yes would be incredibly helpful. Also for trying out different models and providers
